src/main/java/com/mraof/minestuck/entity/animation/ActionCooldown.java

```diff
@@ -0,0 +1,23 @@
+package com.mraof.minestuck.entity.animation;
+
+/**
+ * Used to help standardize the method in which action cooldown(how long between uses of an animated action) values are maintained
+ */
+public interface ActionCooldown
+{
+	//TODO Figure out if it can be made so that group cooldowns can be applied to any animated entities with the same Goal, as opposed to giving a cooldown to just entities of the same type
```

Comment:
You could create a new interface that is implemented by all entites that should have their cooldown reset